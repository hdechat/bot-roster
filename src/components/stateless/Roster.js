import React from 'react';
import TableByCategory from './TableByCategory';

function Roster({ team, updateName, deleteRobot, inRosters }) {
  const { teamName, starters, subs } = team;

  return (
    <div className="Roster">
      <h1>{ teamName }</h1>
      <TableByCategory 
        header="Starters" 
        robots={starters} 
        updateName={updateName} 
        deleteRobot={deleteRobot}
        inRosters={inRosters}/>
      <TableByCategory 
        header="Subs" 
        robots={subs} 
        updateName={updateName} 
        deleteRobot={deleteRobot}
        inRosters={inRosters}/>
    </div>
  );
};

export default Roster;
