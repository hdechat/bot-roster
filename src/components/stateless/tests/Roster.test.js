import React from 'react';
import Roster from '../Roster';
import { shallow } from 'enzyme';

describe('Roster', () => {
  let wrapper;
  let mockTeamName;
  let mockStarters;
  let mockSubs;
  let mockTeam;

  beforeEach(() => {
    mockTeamName = 'Droids';
    mockStarters = [
      {
        id: 'ABC123',
        firstName: 'Twikki',
        lastName: 'Rogers',
        speed: 20,
        strength: 30,
        agility: 40,
        totalAttrScore: 90
      }
    ];
    mockSubs = [
      {
        id: 'XXY678',
        firstName: 'Voltron',
        lastName: 'Smith',
        speed: 20,
        strength: 35,
        agility: 40,
        totalAttrScore: 95
      }
    ];

    mockTeam = {
      teamName: mockTeamName,
      starters: mockStarters,
      subs: mockSubs
    }

    wrapper = shallow(<Roster team={mockTeam}/>)
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the expected elements', () => {
    expect(wrapper.find('div.Roster').length).toEqual(1);
    expect(wrapper.find('h1').length).toEqual(1);
    expect (wrapper.find('TableByCategory').length).toEqual(2);
  });

  it('renders the teamName', () => {
    expect(wrapper.find('h1').text()).toBe(mockTeamName);
  });
});
