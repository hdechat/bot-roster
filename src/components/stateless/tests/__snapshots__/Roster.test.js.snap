// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Roster matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Roster
    team={
        Object {
            "starters": Array [
              Object {
                "agility": 40,
                "firstName": "Twikki",
                "id": "ABC123",
                "lastName": "Rogers",
                "speed": 20,
                "strength": 30,
                "totalAttrScore": 90,
              },
            ],
            "subs": Array [
              Object {
                "agility": 40,
                "firstName": "Voltron",
                "id": "XXY678",
                "lastName": "Smith",
                "speed": 20,
                "strength": 35,
                "totalAttrScore": 95,
              },
            ],
            "teamName": "Droids",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Droids
</h1>,
        <TableByCategory
          deleteRobot={undefined}
          header="Starters"
          inRosters={undefined}
          robots={
                    Array [
                              Object {
                                "agility": 40,
                                "firstName": "Twikki",
                                "id": "ABC123",
                                "lastName": "Rogers",
                                "speed": 20,
                                "strength": 30,
                                "totalAttrScore": 90,
                              },
                            ]
          }
          updateName={undefined}
/>,
        <TableByCategory
          deleteRobot={undefined}
          header="Subs"
          inRosters={undefined}
          robots={
                    Array [
                              Object {
                                "agility": 40,
                                "firstName": "Voltron",
                                "id": "XXY678",
                                "lastName": "Smith",
                                "speed": 20,
                                "strength": 35,
                                "totalAttrScore": 95,
                              },
                            ]
          }
          updateName={undefined}
/>,
      ],
      "className": "Roster",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Droids",
        },
        "ref": null,
        "rendered": "Droids",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "deleteRobot": undefined,
          "header": "Starters",
          "inRosters": undefined,
          "robots": Array [
            Object {
              "agility": 40,
              "firstName": "Twikki",
              "id": "ABC123",
              "lastName": "Rogers",
              "speed": 20,
              "strength": 30,
              "totalAttrScore": 90,
            },
          ],
          "updateName": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "deleteRobot": undefined,
          "header": "Subs",
          "inRosters": undefined,
          "robots": Array [
            Object {
              "agility": 40,
              "firstName": "Voltron",
              "id": "XXY678",
              "lastName": "Smith",
              "speed": 20,
              "strength": 35,
              "totalAttrScore": 95,
            },
          ],
          "updateName": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Droids
</h1>,
          <TableByCategory
            deleteRobot={undefined}
            header="Starters"
            inRosters={undefined}
            robots={
                        Array [
                                    Object {
                                      "agility": 40,
                                      "firstName": "Twikki",
                                      "id": "ABC123",
                                      "lastName": "Rogers",
                                      "speed": 20,
                                      "strength": 30,
                                      "totalAttrScore": 90,
                                    },
                                  ]
            }
            updateName={undefined}
/>,
          <TableByCategory
            deleteRobot={undefined}
            header="Subs"
            inRosters={undefined}
            robots={
                        Array [
                                    Object {
                                      "agility": 40,
                                      "firstName": "Voltron",
                                      "id": "XXY678",
                                      "lastName": "Smith",
                                      "speed": 20,
                                      "strength": 35,
                                      "totalAttrScore": 95,
                                    },
                                  ]
            }
            updateName={undefined}
/>,
        ],
        "className": "Roster",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Droids",
          },
          "ref": null,
          "rendered": "Droids",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "deleteRobot": undefined,
            "header": "Starters",
            "inRosters": undefined,
            "robots": Array [
              Object {
                "agility": 40,
                "firstName": "Twikki",
                "id": "ABC123",
                "lastName": "Rogers",
                "speed": 20,
                "strength": 30,
                "totalAttrScore": 90,
              },
            ],
            "updateName": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "deleteRobot": undefined,
            "header": "Subs",
            "inRosters": undefined,
            "robots": Array [
              Object {
                "agility": 40,
                "firstName": "Voltron",
                "id": "XXY678",
                "lastName": "Smith",
                "speed": 20,
                "strength": 35,
                "totalAttrScore": 95,
              },
            ],
            "updateName": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
